### PR TITLE
Blue Mage corrections/additions

### PR DESCRIPTION
**Pull request by xiashtra**
Song of Torment, Aetherial Spark, and Nightbloom all give the 6B2 Bleeding status with different durations, and possibly different potencies (depending on if Bristle is used on Spark and Torment).  They will overwrite each other.

Stotram has two different ActionIDs for when it's used as a heal vs damage spell.

Condensed Libra gives one of three possible debuffs:
  849 Astral Attenuation: 5% increased damage from fire, wind, or lightning damage
  84A Umbral Attenuation: 5% increased damage from water, earth, or ice damage
  84B Physical Attenuation: 5% increased damage from physical damage
How to handle the Astral/Umbral Attenuations?  Are damage subtypes [i.e. Magic (Fire)] valid inputs?

### CODE REVIEW

File: Definitions/BlueMage.json [hunk 1]
```diff
@@ -6,37 +6,39 @@
...
+{ "2C7E" : "aqua breath", "damage": [ { "potency": 70}, { "potency": 140, "targetindex": 0} ] },
 { "2C7F" : "plaincracker", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "2C82" : "mind blast", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
 { "2C83" : "blood drain", "damage": [ { "potency": 50} ] },
 { "2C84" : "bomb toss", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
 { "2C85" : "1000 needles", "damage": [ { "potency": 0} ] },
 { "2C86" : "drill cannons", "damage": [ { "potency": 0} ] },
 { "2C87" : "the look", "damage": [ { "potency": 110 }, { "potency": 220, "targetindex": 0} ] },
-{ "2C88" : "sharpened knife", "damage": [ { "potency": 220} ] },
+{ "2C88" : "sharpened knife", "damage": [ { "potency": 0} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:17]: potency can increase from 220 to 450 if the target is stunned.

File: Definitions/BlueMage.json [hunk 2]
```diff
@@ -6,37 +6,39 @@
...
 { "2C8A" : "flame thrower", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "2C8C" : "glower", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "2C8D" : "missile", "damage": [ { "potency": 0} ] },
 { "2C8E" : "white wind", "heal": [ { "potency": 0} ] },
-{ "2C8F" : "final sting", "damage": [ { "potency": 1500} ] },
-{ "2C90" : "self-destruct", "damage": [ { "potency": 1500} ] },
+{ "2C8F" : "final sting", "damage": [ { "potency": 2000} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:22]: buffed from 1500 to 2000 in 5.1

File: Definitions/BlueMage.json [hunk 3]
```diff
@@ -6,37 +6,39 @@
...
+{ "2C90" : "self-destruct", "damage": [ { "potency": 0} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:23]: potency can increase from 1500 to 1800 if caster has the Toad Oil buff

File: Definitions/BlueMage.json [hunk 4]
```diff
@@ -6,37 +6,39 @@
 { "2C7A" : "song of torment", "damage": [ { "potency": 50} ] },
 { "2C7B" : "high voltage", "damage": [ { "potency": 0} ] },
 { "2C7D" : "flying frenzy", "damage": [ { "potency": 75}, { "potency": 150, "targetindex": 0} ] },
+{ "2C7E" : "aqua breath", "damage": [ { "potency": 70}, { "potency": 140, "targetindex": 0} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:9]: formatting

File: Definitions/BlueMage.json [hunk 5]
```diff
@@ -6,37 +6,39 @@
...
 { "2C91" : "transfusion", "heal": [ { "potency": 0} ] },
 { "2C95" : "tail screw", "damage": [ { "potency": 0} ] },
 { "2C9B" : "the ram's voice", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "2C9C" : "the dragon's voice", "damage": [ { "potency": 0} ] },
 { "2C9E" : "ink jet", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
 { "2C9F" : "flying sardine", "damage": [ { "potency": 10} ] },
 { "2CA1" : "fire angon", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
-{ "2CA2" : "feather rain", "damage": [ { "potency": 240} ] },
+{ "2CA2" : "feather rain", "damage": [ { "potency": 220} ] },
 { "2CA3" : "eruption", "damage": [ { "potency": 300} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:32]: incorrect initial hit potency

File: Definitions/BlueMage.json [hunk 6]
```diff
@@ -6,37 +6,39 @@
...
 { "2CA4" : "mountain buster 2ca4", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
 { "2CA5" : "shock strike", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
 { "2CA6" : "glass dance", "damage": [ { "potency": 175}, { "potency": 350, "targetindex": 0} ] },
 { "4777" : "alpine draft", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "4778" : "protean wave", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
+{ "4779" : "Northerlies", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "477A" : "electrogenesis", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "477B" : "kaltstrahl", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
-{ "477C" : "abyssal transfiction", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
+{ "477C" : "Abyssal Transfixion", "damage": [ { "potency": 220} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:41]: single target attack not AoE

File: Definitions/BlueMage.json [hunk 7]
```diff
@@ -59,7 +61,7 @@
 { "5AE6" : "saintly beam", "damage": [ { "potency": 0} ] },
 { "5AE7" : "feculent flood", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "5AE8" : "angel's snack", "heal": [ { "potency": 400} ] },
-{ "5AEA" : "divine cataract", "damage": [ { "potency": 250}, { "potency": 500, "targetindex": 0} ] },
+{ "5AEA" : "divine cataract", "damage": [ { "potency": 0} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:64]: potency can increase to 1000 while under tank mimicry

File: Definitions/BlueMage.json [hunk 8]
```diff
@@ -72,10 +74,13 @@
 { "5AF7" : "both ends", "damage": [ { "potency": 600} ] },
 { "5AF9" : "phantom flurry", "damage": [ { "potency": 300}, {"potency":600, "targetindex": 0} ] },
 { "5AFA" : "nightbloom", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
-{ "2C7E" : "aqua breath", "damage": [ { "potency": 70}, { "potency": 140, "targetindex": 0} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:75]: formatting

File: Definitions/BlueMage.json [hunk 9]
```diff
@@ -72,10 +74,13 @@
...
 { "5AFA" : "nightbloom", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
-{ "2C7E" : "aqua breath", "damage": [ { "potency": 70}, { "potency": 140, "targetindex": 0} ] },
+{ "5B78" : "stotram", "damage": [ { "potency": 140} ], "heal": [ { "potency": 300} ] },
 ],
 "statuseffects" : [ 
-    { "84D": "aetherial mimicry: dps", "multiplier": [ { "type": "criticalhit", "amount": 20 } ] },
+{ "84D": "aetherial mimicry: dps", "multiplier": [
+    { "type": "criticalhit", "amount": 20 },
+    { "type": "directhit", "amount": 20 }
+] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:83]: dps mimicry increases both critical and direct hit rate

File: Definitions/BlueMage.json [hunk 10]
```diff
@@ -72,10 +74,13 @@
 { "5AF7" : "both ends", "damage": [ { "potency": 600} ] },
 { "5AF9" : "phantom flurry", "damage": [ { "potency": 300}, {"potency":600, "targetindex": 0} ] },
 { "5AFA" : "nightbloom", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
-{ "2C7E" : "aqua breath", "damage": [ { "potency": 70}, { "potency": 140, "targetindex": 0} ] },
+{ "5B78" : "stotram", "damage": [ { "potency": 140} ], "heal": [ { "potency": 300} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:77]: Stotram has separate actionIDs when used as a heal vs damage spell

File: Definitions/BlueMage.json (hunk 8 above)
Comment (xiashtra) [inline Definitions/BlueMage.json:75]: formatting (moved to top to maintain ID ordering)

File: Definitions/BlueMage.json [hunk 11]
```diff
@@ -84,7 +89,7 @@
 { "6B2": "nightbloom", "timeproc": { "type": "dot", "potency": "75", "maxticks": "20" } },
 { "6C8": "dropsy", "timeproc": { "type": "dot", "potency": "20", "damagetype": "magic", "maxticks": "4" } },
 
-{ "6B4" : "boost", "potency": [ {"type": "damagedonemultiplier", "amount": 50} ] },
+{ "6B4" : "boost", "potency": [ {"type": "damagedonemultiplier", "amount": 50, "limittoactioncategory": "spell"} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:92]: Bristle only effects spells and not abilities

File: Definitions/BlueMage.json [hunk 12]
```diff
@@ -93,16 +98,22 @@
 ] },
 { "6B9" : "peculiar light", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5, "limittodamagetype": "Magic"} ] },
 { "6BA" : "diamondback", "potency": [ {"type": "DamageReceivedMultiplier", "amount": -90} ] },
-{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical"} ] },
-{ "847" : "cactguard", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5} ] },
-{ "849" : "physical attenuation", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5} ] },
+{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical", "limittoactioncategory": "spell"} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:101]: Whistle only effects spells and not abilities

File: Definitions/BlueMage.json [hunk 13]
```diff
@@ -93,16 +98,22 @@
 ] },
 { "6B9" : "peculiar light", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5, "limittodamagetype": "Magic"} ] },
 { "6BA" : "diamondback", "potency": [ {"type": "DamageReceivedMultiplier", "amount": -90} ] },
-{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical"} ] },
-{ "847" : "cactguard", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5} ] },
-{ "849" : "physical attenuation", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5} ] },
+{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical", "limittoactioncategory": "spell"} ] },
+{ "847" : "cactguard", "potency": [ {"type": "DamageReceivedMultiplier", "amount": -5} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:102]: Cactguard multiplier had incorrect sign

File: Definitions/BlueMage.json [hunk 14]
```diff
@@ -93,16 +98,22 @@
 ] },
 { "6B9" : "peculiar light", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5, "limittodamagetype": "Magic"} ] },
 { "6BA" : "diamondback", "potency": [ {"type": "DamageReceivedMultiplier", "amount": -90} ] },
-{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical"} ] },
-{ "847" : "cactguard", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5} ] },
-{ "849" : "physical attenuation", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5} ] },
+{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical", "limittoactioncategory": "spell"} ] },
+{ "847" : "cactguard", "potency": [ {"type": "DamageReceivedMultiplier", "amount": -5} ] },
+{ "84B" : "physical attenuation", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5, "limittodamagetype": "Physical"} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:103]: Condensed Libra: Physical Attenuation is 84B

849 is Astral Attenuation and 84A is Umbral Attenuation

File: Definitions/BlueMage.json [hunk 15]
```diff
@@ -93,16 +98,22 @@
...
+{ "846" : "harmonized", "potency": [ {"type": "damagedonemultiplier", "amount": 80, "limittodamagetype": "Physical", "limittoactioncategory": "spell"} ] },
+{ "847" : "cactguard", "potency": [ {"type": "DamageReceivedMultiplier", "amount": -5} ] },
+{ "84B" : "physical attenuation", "potency": [ {"type": "DamageReceivedMultiplier", "amount": 5, "limittodamagetype": "Physical"} ] },
 { "84E" : "aetherial mimicry: healer", "potency": [ {"type": "HealDoneMultiplier", "amount": 20} ] },
 { "9BC" : "tingling", "potency": [ {"type": "DamageAddPotency", "amount": 100, "limittodamagetype": "Physical"} ] },
 
-{ "6B8": "ice spikes", "reactiveproc": { "type": "DamageOnDamageReceived" } },
-{ "6BC": "veil of the whorl", "reactiveproc": { "type": "DamageOnDamageReceived" } },
+{ "6B8": "ice spikes", "reactiveproc": { "type": "DamageOnDamageReceived", "amount": 40 } },
+{ "6BC": "veil of the whorl", "reactiveproc": { "type": "DamageOnDamageReceived", "amount": 50 } },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:108]: added potency of counter damage for Ice Spikes and Veil of the Whorl

File: Definitions/BlueMage.json [hunk 16]
```diff
@@ -93,16 +98,22 @@
...
 
 { "852" : "Surpanakha's Fury", "potency": [ {"type": "damagedonemultiplier", "amount": 50, "isstacked": true, "limitto": "Surpanakha"} ] },
+    
+{ "6B3" : "Bad Breath", "potency": [ {"type": "damagedonemultiplier", "amount": -10} ] },
+{ "12"  : "Bad Breath", "timeproc": { "type": "dot", "potency": "20", "damagetype": "magic", "maxticks": "5" } },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:113]: Bad Breath gives up to six different status effects:

09 Slow
11 Paralysis
12 Poison (damage over time, 20 potency for 15s)
0E Heavy
0F Blind
6B3 Malodorus (damage dealt down 10%)

File: Definitions/BlueMage.json [hunk 17]
```diff
@@ -93,16 +98,22 @@
...
+{ "842" : "Gobskin", "damageshield": { "type": "Potency", "amount": 0 } },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:114]: Gobskin shield potency amount is dependent on mimicry

File: Definitions/BlueMage.json [hunk 18]
```diff
@@ -93,16 +98,22 @@
...
+{ "843" : "Magic Hammer", "potency": [ { "type": "damagedonemultiplier", "amount": -10, "limittodamagetype": "magic"} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:115]: Magic Hammer gives the 843 Conked debuff which reduces intelligence and mind by 10%

File: Definitions/BlueMage.json [hunk 19]
```diff
@@ -93,16 +98,22 @@
...
+{ "9C0" : "Chelonian Gate", "potency": [ { "type": "DamageReceivedMultiplier", "amount": -20 }] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:116]: Chelonian Gate reduces damage taken by 20% while channeled

File: Definitions/BlueMage.json [hunk 20]
```diff
@@ -6,37 +6,39 @@
...
 { "2CA1" : "fire angon", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
-{ "2CA2" : "feather rain", "damage": [ { "potency": 240} ] },
+{ "2CA2" : "feather rain", "damage": [ { "potency": 220} ] },
 { "2CA3" : "eruption", "damage": [ { "potency": 300} ] },
 { "2CA4" : "mountain buster 2ca4", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
 { "2CA5" : "shock strike", "damage": [ { "potency": 200}, { "potency": 400, "targetindex": 0} ] },
 { "2CA6" : "glass dance", "damage": [ { "potency": 175}, { "potency": 350, "targetindex": 0} ] },
 { "4777" : "alpine draft", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "4778" : "protean wave", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
+{ "4779" : "Northerlies", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:38]: Northerlies was missing

File: Definitions/BlueMage.json [hunk 21]
```diff
@@ -6,37 +6,39 @@
...
+{ "2C90" : "self-destruct", "damage": [ { "potency": 0} ] },
 { "2C91" : "transfusion", "heal": [ { "potency": 0} ] },
 { "2C95" : "tail screw", "damage": [ { "potency": 0} ] },
 { "2C9B" : "the ram's voice", "damage": [ { "potency": 110}, { "potency": 220, "targetindex": 0} ] },
 { "2C9C" : "the dragon's voice", "damage": [ { "potency": 0} ] },
 { "2C9E" : "ink jet", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
 { "2C9F" : "flying sardine", "damage": [ { "potency": 10} ] },
 { "2CA1" : "fire angon", "damage": [ { "potency": 100}, { "potency": 200, "targetindex": 0} ] },
-{ "2CA2" : "feather rain", "damage": [ { "potency": 240} ] },
+{ "2CA2" : "feather rain", "damage": [ { "potency": 220} ] },
```
Comment (xiashtra) [inline Definitions/BlueMage.json:31]: incorrect initial hit potency

**Comment by ravahn** [comment]
This looks great, thanks for the help correcting and adding Blu skills.  Merging to the repository to record your contribution.  This will be included in the next plugin update.

**Comment by blacc000** [comment]
您的邮件我已收到。

**Pull request merged**